tests/test_bidstream_client.py

```diff
@@ -0,0 +1,160 @@
+import unittest
+from unittest.mock import patch
+
+from test_utils import *
+from uid2_client import BidstreamClient, EncryptionError, Uid2Base64UrlCoder
+
+
+@patch('uid2_client.bid_stream_client.refresh_bidstream_keys')
+class TestBidStreamClient(unittest.TestCase):
+    _CONST_BASE_URL = 'base_url'
+    _CONST_API_KEY = 'api_key'
+
+    def setUp(self):
+        self._key_collection = create_key_collection(IdentityScope.UID2)
+        self._client = BidstreamClient(self._CONST_BASE_URL, self._CONST_API_KEY, client_secret)
+
+    def test_smoke_test(self, mock_refresh_bidstream_keys):  # SmokeTest
+        for expected_scope, expected_version in test_cases_all_scopes_all_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                token = generate_uid_token(expected_scope, expected_version)
+                mock_refresh_bidstream_keys.return_value = create_key_collection(expected_scope)
+                self._client.refresh_keys()
+                decrypted = self._client.decrypt_token_into_raw_uid(token, None)
+                self.assertEqual(decrypted.identity_scope, expected_scope)
+                self.assertEqual(decrypted.advertising_token_version, expected_version)
+                self.assertEqual((now - decrypted.established).total_seconds(), 0)
```

Comment:
where do we do the equivalent of `Assert.Equal(EXAMPLE_EMAIL_RAW_UID2_V2, res.Uid);` ?
Can we use a `DecryptAndAssertSuccess` equivalent for the 3 relevant tests?